ipaclient/plugins/vault.py

```diff
@@ -813,7 +861,46 @@ def forward(self, *args, **options):
 
         options['vault_data'] = wrapped_vault_data
 
-        return self.api.Command.vault_archive_internal(*args, **options)
+        for retry in (True, False):
+            # retrieve transport certificate
+            try:
+                transport_cert = get_cached_transport_cert(api.env.domain)
+            except KeyError:
+                config = self.api.Command.vaultconfig_show()['result']
+                transport_cert_der = config['transport_cert']
+                transport_cert = (
+                    x509.load_certificate(transport_cert_der, x509.DER))
+                set_cached_transport_cert(api.env.domain, transport_cert)
+            else:
+                transport_cert_der = (
+                    transport_cert.public_bytes(serialization.Encoding.DER))
```

Comment:
Please move this functionality to a plugin or some other place. I like to fetch and cache the KRA transport cert every time Custodia is started. The API should have an optional argument to force a refresh.